### PR TITLE
Golang assignment 6

### PR DESCRIPTION
**Pull request by divyatambat**
1. In the given code, the accessSlice function accepts a slice and index. 
If the value is present in the slice at that index, the program should print the following.

"item <index>, value <value at that index in slice>"

But if the index does not hold any value,
it will lead to an index out of range panic in our program.

Complete the given code to recover from panic inside the accessSlice function, when index out of range panic occurs.
Also, Print the following after handling panic

"internal error: <recovered panic value>"

Example Test Case 1 :

Input: 3
Output:
item 3, value 6
Testing Panic and Recover

2. In the given code, the accessSlice function accepts a slice and index.
If the value is present in the slice at that index, the program should print the following.

"item <index>, value <value present at the index>"

But if the index does not hold any value,
it will lead to an index out of range panic in our program.

So in order to safeguard our program from panicking, add a condition to handle the condition if 

index > lengthOfSlice - 1

and return an error from the accessSlice function if the above condition is met.
The error message should be the following :

"length of the slice should be more than index"

Complete the given program to return an error from the accessSlice function and handle the returned error inside the main function to print the message. 

Example Test Case 1 :

Input: 3
Output:
item 3, value 6

### CODE REVIEW

File: Asg0.6-1.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package main
+
+import "fmt"
+
+func accessSlice(slice []int, index int) {
+
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Printf("Internal error: %d\n", err)
```
Comment (yama11299) [inline Asg0.6-1.go]: why %d? to print the error use %s.

File: Asg0.6-2.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+func main() {
+	slice := []int{10, 20, 30, 40, 50}
+	var index int
+	fmt.Print("Input: ")
+	fmt.Scanln(&index)
+
+	value, err := accessSlice(slice, index)
+	if err != nil {
+		fmt.Printf("Internal error: %d\n", err)
```
Comment (yama11299) [inline Asg0.6-2.go]: follow Go's error handling standard. return from here itself.

File: asg0.4-2.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+func main() {
+	var length, width int
+
+	fmt.Print("Enter length of rectangle: ")
+	fmt.Scan(&length)
+	fmt.Print("Enter width of rectangle: ")
+	fmt.Scan(&width)
+
+	calculationOfRectangle := RectangleAsg2{Length: length, Width: width}
```
Comment (yama11299) [inline asg0.4-2.go:29]: why such variable name?

File: Asg0.6-1.go (hunk 1 above)
Comment (divyatambat) [reply Asg0.6-1.go]: done

File: Asg0.6-2.go (hunk 2 above)
Comment (divyatambat) [reply Asg0.6-2.go]: ok, done

File: asg0.4-2.go (hunk 3 above)
Comment (divyatambat) [reply asg0.4-2.go:29]: variable names changed

**Pull request merged**